File: src/main/java/adventOfCode/lib/SolverMap.java
package adventOfCode.lib;

import java.util.HashMap;
import java.util.Map;

public class SolverMap {
    private static final Map<String, Solver> SOLVERS = new HashMap<>();

    static {
        SOLVERS.put("2023/01", new adventOfCode.y2023.Day01());
    }

    public static Solver getSolver(String name) {
        return SOLVERS.get(name);
    }
}

File: src/main/kotlin/adventOfCode/lib/SolverMap.kt
package adventOfCode.lib

object SolverMap {
    private val solvers: Map<String, Solver> = hashMapOf(
        "2015/01" to adventOfCode.y2015.d01.Solution(),
        "2015/02" to adventOfCode.y2015.d02.Solution(),
        "2023/01" to adventOfCode.y2023.d01.Solution()
    )

    fun getSolver(name: String): Solver? {
        return solvers[name]
    }
}
